### PR TITLE
[SPARK-46914][UI] Shorten app name in the summary table on the History Page

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This Pull Request shortens long app names to prevent overflow in the app table.

### Why are the changes needed?

better UX

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

new js tests and built and tested locally:
image
image

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @yaooqinn .

File: core/src/main/resources/org/apache/spark/ui/static/utils.js [hunk 1]
```diff
@@ -272,3 +272,16 @@ function errorMessageCell(errorMessage) {
   const details = detailsUINode(isMultiline, errorMessage);
   return summary + details;
 }
+
+function stringAbbreviate(content, limit) {
+  if (content && content.length > limit) {
+    const summary = content.substring(0, limit) + '...';
+    // TODO: Reused stacktrace-details* style for convenience, but it's not really a stacktrace
```
Comment (dongjoon-hyun) [inline core/src/main/resources/org/apache/spark/ui/static/utils.js:279]: Please use IDed TODO with JIRA ID next time~

Comment (dongjoon-hyun) [reply core/src/main/resources/org/apache/spark/ui/static/utils.js:279]: Then, could you avoid to use `TODO`? TODO is a recognized keyword in the IDE.

Comment (yaooqinn) [reply core/src/main/resources/org/apache/spark/ui/static/utils.js:279]: OK, I got it. Thank you for the explanation, @dongjoon-hyun

**Pull request closed without merging**